modules/caddyhttp/celmatcher.go

```diff
@@ -137,14 +165,15 @@ func (m *MatchExpression) Provision(_ caddy.Context) error {
 
 // Match returns true if r matches m.
 func (m MatchExpression) Match(r *http.Request) bool {
-	out, _, _ := m.prg.Eval(map[string]interface{}{
-		"request": celHTTPRequest{r},
-	})
+	celReq := celHTTPRequest{r}
+	out, _, err := m.prg.Eval(celReq)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
Hmm, I definitely don't think we should `panic`. Caddy _must_ keep running.

We added a mechanism to bubble up an error from within a matcher a while back, so instead do this (and return false):

```go
caddyhttp.SetVar(r.Context(), caddyhttp.MatcherErrorVarKey, err)
```

The godoc:
```go
// MatcherErrorVarKey is the key used for the variable that
// holds an optional error emitted from a request matcher,
// to short-circuit the handler chain, since matchers cannot
// return errors via the RequestMatcher interface.
```

Reply:
I totally forgot about this 😶

Reply:
Fixed.